ng/src/web/pages/usersettings/usersettingspage.js

```diff
@@ -0,0 +1,1064 @@
+    this.getLanguageNameByCode = this.getLanguageNameByCode.bind(this);
+    this.getFilterNameById = this.getFilterNameById.bind(this);
+    this.getValueBySettingId = this.getValueBySettingId.bind(this);
+
+    this.openSettingsDialog = this.openSettingsDialog.bind(this);
+    this.handleSaveSettings = this.handleSaveSettings.bind(this);
+    this.handleCloseDialog = this.handleCloseDialog.bind(this);
+  }
+
+  componentDidMount() {
+    this.load();
+  }
+
+  load() {
+    this.readSettings()
+    .then(response => this.createSettinggroupTables(response))
+    .then(() => this.setState({loading: false}));
+  }
+
+  readSettings() {
+    const {gmp} = this.props;
+    const {option_lists} = this.state;
+    const all_settings = Promise.all([
+      gmp.user.currentSettings()
+      .then(response => {
+        return response.data.getEntries();
+      }),
+      gmp.filters.getAll()
+      .then(response => {
+        return response.getEntries();
+      }),
+      gmp.scanconfigs.getAll()
+      .then(response => {
+        const config_list = response.getEntries();
+        option_lists.scanconfigsList = config_list;
+        this.setState({option_lists});
+        return config_list;
+      }),
+      gmp.user.currentLanguages()
+      .then(response => {
+        option_lists.languagesList = response.data;
+        this.setState({option_lists});
+        return option_lists.languagesList;
+      }),
+      gmp.alerts.getAll()
+      .then(response => {
+        option_lists.alertsList = response.getEntries();
+        this.setState({option_lists});
+      }),
+      gmp.credentials.getAll()
+      .then(response => {
+        option_lists.credentialsList = response.getEntries();
+        this.setState({option_lists});
+      }),
+      gmp.portlists.getAll()
+      .then(response => {
+        option_lists.portlistsList = response.getEntries();
+        this.setState({option_lists});
+      }),
+      gmp.reportformats.getAll()
+      .then(response => {
+        option_lists.reportformatsList = response.getEntries();
+        this.setState({option_lists});
+      }),
+      gmp.scanners.getAll()
+      .then(response => {
+        option_lists.scannersList = response.getEntries();
+        this.setState({option_lists});
+      }),
+      gmp.reportformats.getAll()
+      .then(response => {
+        option_lists.reportformatsList = response.getEntries();
+        this.setState({option_lists});
+      }),
+      gmp.schedules.getAll()
+      .then(response => {
+        option_lists.schedulesList = response.getEntries();
+        this.setState({option_lists});
+      }),
+      gmp.targets.getAll()
+      .then(response => {
+        option_lists.targetsList = response.getEntries();
+        this.setState({option_lists});
+      }),
+      gmp.filters.getAll()
+      .then(response => {
+        const filtersList = response.getEntries();
+        const filtersListAgent = filtersList.filter(
+          item => {
```

Comment:
The filters arrow functions could have been written shorter `filter(item => item.filter_type === 'Xyz')`